### PR TITLE
[GSoC] physics/continuum_mechanics: solve Method along with documentation and tests added for the Truss class 

### PR DESCRIPTION
**Pull request by AdvaitPote**
#### Brief description of what is fixed or changed
After the implementation of the `truss` class with the initialising methods like `add_node` and `add_method` along with many others, the main solving method responsible for solving a given Truss added. After calling the `solve` method for a given truss, the `reaction_loads` and `internal_forces` are updated with the calculated values. 

#### Release Notes

* physics.continuum_mechanics
  * Added the `solve` method for solving a given `truss` for its reaction loads and internal forces.

### CODE REVIEW

**Comment by AdvaitPote** [comment]
@Psycho-Pirate

File: sympy/physics/continuum_mechanics/truss.py [hunk 1]
```diff
@@ -599,3 +608,108 @@ def remove_support(self, location):
...
+        Where n is the number of nodes, r is the number of reaction forces, where each pinned
+        support has 2 reaction forces and each roller has 1, and m is the number of members.
+
+        The given condition is derived from the fact that a system of equations is solvable
+        only when the number of variables is lesser than or equal to the number of equations.
+        Equilibrium Equations in x and y directions give two equations per node giving 2n number
+        equations. However, the truss needs to be stable as well and may be unstable if 2n > r + m.
+        The number of variables is simply the sum of the number of reaction forces and member
+        forces.
+
```
Comment (Psycho-Pirate) [inline sympy/physics/continuum_mechanics/truss.py:630]: Please mention what signs the compressive and tensile forces are using.

**Comment by Psycho-Pirate** [comment]
I tried solving the following problem
image

Here is the code:
```
from sympy.physics.continuum_mechanics.truss import Truss
import math
t = Truss()

>>> t.add_node("A", -4, 0)
>>> t.add_node("B", 0, 0)
>>> t.add_node("C", 4, 0)
>>> t.add_node("D", 8, 0)
>>> t.add_node("E", 6, 2/math.sqrt(3))
>>> t.add_node("F", 2, 2*math.sqrt(3))
>>> t.add_node("G", -2, 2/math.sqrt(3))

>>> t.add_member("AB","A","B")
>>> t.add_member("BC","B","C")
>>> t.add_member("CD","C","D")
>>> t.add_member("AG","A","G")
>>> t.add_member("GB","G","B")
>>> t.add_member("GF","G","F")
>>> t.add_member("BF","B","F")
>>> t.add_member("FC","F","C")
>>> t.add_member("CE","C","E")
>>> t.add_member("FE","F","E")
>>> t.add_member("DE","D","E")

>>> t.apply_support("A","pinned")
>>> t.apply_support("D","roller")
>>> t.apply_load("G", 3, 90)
>>> t.apply_load("E", 3, 90)
>>> t.apply_load("F", 2, 90)
>>> t.solve()
>>> t.reaction_loads
{'R_A_x': 3.98027755288532e-16, 'R_A_y': -4.00000000000000, 'R_D_y': -4.00000000000000}
```
It solves the internal forces and the answers are matching but for reaction forces, `R_A_x` should equal to zero,

**Comment by AdvaitPote** [comment]
@Psycho-Pirate I have made the required changes. For the second issue, there wasn't any major flaw with how equations were solved but just that when dealing with floats, it just returned a term tending to an infinitesimal term. To get around this, I just found out the minimum load applied and then iterated through the `forces_matrix` and if there happened to be a force which is smaller than `min_load` by a certain degree, `1E-10` in this case, then that force would just be reduced to zero.

**Comment by AdvaitPote** [comment]
@Psycho-Pirate

**Comment by Psycho-Pirate** [comment]
Looks good to me! Merging!

**Pull request merged**